// mongodb-stitch/packages/core/admin-client/dist/esm/authProviders/ProviderConfigs.js
var ProviderConfigCodec = (function () {
    function ProviderConfigCodec() {
    }
    ProviderConfigCodec.prototype.decode = function (from) {
        return {
            config: from.config,
            type: from.type
        };
    };
    ProviderConfigCodec.prototype.encode = function (from) {
        return {
            config: from.config,
            type: from.type
        };
    };
    return ProviderConfigCodec;
}());
export { ProviderConfigCodec };
var Anon = (function () {
    function Anon() {
        this.type = "anon-user";
    }
    return Anon;
}());
export { Anon };
var ApiKey = (function () {
    function ApiKey() {
    }
    return ApiKey;
}());
export { ApiKey };
var Userpass = (function () {
    function Userpass(emailConfirmationUrl, resetPasswordUrl, confirmEmailSubject, resetPasswordSubject) {
        this.emailConfirmationUrl = emailConfirmationUrl;
        this.resetPasswordUrl = resetPasswordUrl;
        this.confirmEmailSubject = confirmEmailSubject;
        this.resetPasswordSubject = resetPasswordSubject;
        this.type = "local-userpass";
        this.config = {
            confirmEmailSubject: this.confirmEmailSubject,
            emailConfirmationUrl: this.emailConfirmationUrl,
            resetPasswordSubject: this.resetPasswordSubject,
            resetPasswordUrl: this.resetPasswordUrl
        };
    }
    return Userpass;
}());
export { Userpass };
var Custom = (function () {
    function Custom(signingKey) {
        this.signingKey = signingKey;
        this.type = "custom-token";
        this.config = {
            signingKey: this.signingKey
        };
    }
    return Custom;
}());
export { Custom };
