// mongodb-stitch/packages/core/admin-client/dist/umd/index.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./authProviders/ProviderConfigs", "./Resources", "./services/rules/RulesResources", "./services/ServiceConfigs", "./StitchAdminClient"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var ProviderConfigs_1 = require("./authProviders/ProviderConfigs");
    exports.Anon = ProviderConfigs_1.Anon;
    exports.Custom = ProviderConfigs_1.Custom;
    exports.Userpass = ProviderConfigs_1.Userpass;
    var Resources_1 = require("./Resources");
    exports.App = Resources_1.App;
    exports.Apps = Resources_1.Apps;
    exports.AuthProvider = Resources_1.AuthProvider;
    exports.AuthProviders = Resources_1.AuthProviders;
    exports.Function = Resources_1.Function;
    exports.Functions = Resources_1.Functions;
    exports.Rule = Resources_1.Rule;
    exports.Rules = Resources_1.Rules;
    exports.Service = Resources_1.Service;
    exports.Services = Resources_1.Services;
    exports.User = Resources_1.User;
    exports.UserRegistrations = Resources_1.UserRegistrations;
    exports.Users = Resources_1.Users;
    var RulesResources_1 = require("./services/rules/RulesResources");
    exports.AwsRuleCreator = RulesResources_1.AwsRuleCreator;
    exports.AwsS3Actions = RulesResources_1.AwsS3Actions;
    exports.AwsS3RuleCreator = RulesResources_1.AwsS3RuleCreator;
    exports.AwsSesActions = RulesResources_1.AwsSesActions;
    exports.AwsSesRuleCreator = RulesResources_1.AwsSesRuleCreator;
    exports.HttpActions = RulesResources_1.HttpActions;
    exports.HttpRuleCreator = RulesResources_1.HttpRuleCreator;
    exports.MongoDbRuleCreator = RulesResources_1.MongoDbRuleCreator;
    exports.RuleResponse = RulesResources_1.RuleResponse;
    exports.TwilioActions = RulesResources_1.TwilioActions;
    exports.TwilioRuleCreator = RulesResources_1.TwilioRuleCreator;
    var ServiceConfigs_1 = require("./services/ServiceConfigs");
    exports.Aws = ServiceConfigs_1.Aws;
    exports.AwsS3 = ServiceConfigs_1.AwsS3;
    exports.AwsSes = ServiceConfigs_1.AwsSes;
    exports.Http = ServiceConfigs_1.Http;
    exports.Mongo = ServiceConfigs_1.Mongo;
    exports.Twilio = ServiceConfigs_1.Twilio;
    var StitchAdminClient_1 = __importDefault(require("./StitchAdminClient"));
    exports.StitchAdminClient = StitchAdminClient_1.default;
});
