// mongodb-stitch/packages/core/admin-client/dist/umd/users/UsersResources.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var UserCreatorCodec = (function () {
        function UserCreatorCodec() {
        }
        UserCreatorCodec.prototype.decode = function (from) {
            return {
                email: from.email,
                password: from.password
            };
        };
        UserCreatorCodec.prototype.encode = function (from) {
            return {
                email: from.email,
                password: from.password
            };
        };
        return UserCreatorCodec;
    }());
    exports.UserCreatorCodec = UserCreatorCodec;
    var Fields;
    (function (Fields) {
        Fields["Id"] = "_id";
    })(Fields || (Fields = {}));
    var UserResponseCodec = (function () {
        function UserResponseCodec() {
        }
        UserResponseCodec.prototype.decode = function (from) {
            return {
                id: from[Fields.Id]
            };
        };
        UserResponseCodec.prototype.encode = function (from) {
            var _a;
            return _a = {},
                _a[Fields.Id] = from.id,
                _a;
        };
        return UserResponseCodec;
    }());
    exports.UserResponseCodec = UserResponseCodec;
});
