// mongodb-stitch/packages/core/admin-client/dist/umd/userRegistrations/UserRegistrationsResources.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Fields;
    (function (Fields) {
        Fields["Token"] = "token";
        Fields["TokenId"] = "token_id";
    })(Fields || (Fields = {}));
    var ConfirmationEmailCodec = (function () {
        function ConfirmationEmailCodec() {
        }
        ConfirmationEmailCodec.prototype.decode = function (from) {
            return {
                token: from[Fields.Token],
                tokenId: from[Fields.TokenId]
            };
        };
        ConfirmationEmailCodec.prototype.encode = function (from) {
            var _a;
            return _a = {},
                _a[Fields.Token] = from.token,
                _a[Fields.TokenId] = from.tokenId,
                _a;
        };
        return ConfirmationEmailCodec;
    }());
    exports.ConfirmationEmailCodec = ConfirmationEmailCodec;
});
