// mongodb-stitch/packages/core/admin-client/dist/esm/services/rules/RulesResources.js
export var AwsS3Actions;
(function (AwsS3Actions) {
    AwsS3Actions["Put"] = "put";
    AwsS3Actions["SignPolicy"] = "signPolicy";
})(AwsS3Actions || (AwsS3Actions = {}));
export var AwsSesActions;
(function (AwsSesActions) {
    AwsSesActions["Send"] = "send";
})(AwsSesActions || (AwsSesActions = {}));
export var HttpActions;
(function (HttpActions) {
    HttpActions["Get"] = "get";
    HttpActions["Post"] = "post";
    HttpActions["Put"] = "put";
    HttpActions["Delete"] = "delete";
    HttpActions["Head"] = "head";
    HttpActions["Patch"] = "patch";
})(HttpActions || (HttpActions = {}));
export var TwilioActions;
(function (TwilioActions) {
    TwilioActions["Send"] = "send";
})(TwilioActions || (TwilioActions = {}));
var AwsRuleCreator = (function () {
    function AwsRuleCreator(name, actions) {
        this.name = name;
        this.actions = actions;
        this.type = "aws";
    }
    return AwsRuleCreator;
}());
export { AwsRuleCreator };
var AwsS3RuleCreator = (function () {
    function AwsS3RuleCreator(name, actions) {
        this.name = name;
        this.actions = actions;
        this.type = "aws-s3";
    }
    return AwsS3RuleCreator;
}());
export { AwsS3RuleCreator };
var AwsSesRuleCreator = (function () {
    function AwsSesRuleCreator(name, actions) {
        this.name = name;
        this.actions = actions;
        this.type = "aws-ses";
    }
    return AwsSesRuleCreator;
}());
export { AwsSesRuleCreator };
var HttpRuleCreator = (function () {
    function HttpRuleCreator(name, actions) {
        this.name = name;
        this.actions = actions;
        this.type = "http";
    }
    return HttpRuleCreator;
}());
export { HttpRuleCreator };
var MongoDbRuleCreator = (function () {
    function MongoDbRuleCreator(namespace, rule) {
        this.namespace = namespace;
        this.rule = rule;
        this.type = "mongodb";
    }
    return MongoDbRuleCreator;
}());
export { MongoDbRuleCreator };
var TwilioRuleCreator = (function () {
    function TwilioRuleCreator(name, actions) {
        this.name = name;
        this.actions = actions;
        this.type = "twilio";
    }
    return TwilioRuleCreator;
}());
export { TwilioRuleCreator };
var RuleCreatorCodec = (function () {
    function RuleCreatorCodec() {
    }
    RuleCreatorCodec.prototype.encode = function (from) {
        switch (from.type) {
            case "mongodb":
                return new MongoDbCodec().encode(from);
            default:
                return from;
        }
    };
    return RuleCreatorCodec;
}());
export { RuleCreatorCodec };
var MongoDbCodec = (function () {
    function MongoDbCodec() {
    }
    MongoDbCodec.prototype.encode = function (from) {
        from.rule["namespace"] = from.namespace;
        return from.rule;
    };
    return MongoDbCodec;
}());
export { MongoDbCodec };
var RuleResponse = (function () {
    function RuleResponse() {
    }
    return RuleResponse;
}());
export { RuleResponse };
var RuleResponseCodec = (function () {
    function RuleResponseCodec() {
    }
    RuleResponseCodec.prototype.encode = function (from) {
        return {};
    };
    RuleResponseCodec.prototype.decode = function (from) {
        return {};
    };
    return RuleResponseCodec;
}());
export { RuleResponseCodec };
