// mongodb-stitch/packages/core/admin-client/dist/esm/services/ServiceConfigs.js
var ServiceConfigCodec = (function () {
    function ServiceConfigCodec() {
    }
    ServiceConfigCodec.prototype.decode = function (from) {
        var type = from.type;
        var config = from.config;
        if (type === "twilio") {
            config = new TwilioConfigCodec().decode(config);
        }
        return {
            config: config,
            name: from.name,
            type: from.type
        };
    };
    ServiceConfigCodec.prototype.encode = function (from) {
        return {
            config: from.configCodec
                ? from.configCodec.encode(from.config)
                : from.config,
            name: from.name,
            type: from.type
        };
    };
    return ServiceConfigCodec;
}());
export { ServiceConfigCodec };
var Http = (function () {
    function Http(name) {
        this.name = name;
        this.config = {};
        this.type = "http";
    }
    return Http;
}());
export { Http };
var Aws = (function () {
    function Aws(name, config) {
        this.name = name;
        this.config = config;
        this.type = "aws";
    }
    return Aws;
}());
export { Aws };
var AwsS3 = (function () {
    function AwsS3(name, config) {
        this.name = name;
        this.config = config;
        this.type = "aws-s3";
    }
    return AwsS3;
}());
export { AwsS3 };
var AwsSes = (function () {
    function AwsSes(name, config) {
        this.name = name;
        this.config = config;
        this.type = "aws-ses";
    }
    return AwsSes;
}());
export { AwsSes };
export var TwilioConfigFields;
(function (TwilioConfigFields) {
    TwilioConfigFields["AuthToken"] = "auth_token";
    TwilioConfigFields["AccountSid"] = "sid";
})(TwilioConfigFields || (TwilioConfigFields = {}));
var TwilioConfigCodec = (function () {
    function TwilioConfigCodec() {
    }
    TwilioConfigCodec.prototype.decode = function (from) {
        return {
            accountSid: from[TwilioConfigFields.AccountSid],
            authToken: from[TwilioConfigFields.AuthToken]
        };
    };
    TwilioConfigCodec.prototype.encode = function (from) {
        var _a;
        return _a = {},
            _a[TwilioConfigFields.AuthToken] = from.authToken,
            _a[TwilioConfigFields.AccountSid] = from.accountSid,
            _a;
    };
    return TwilioConfigCodec;
}());
export { TwilioConfigCodec };
var Twilio = (function () {
    function Twilio(name, config) {
        this.name = name;
        this.config = config;
        this.configCodec = new TwilioConfigCodec();
        this.type = "twilio";
    }
    return Twilio;
}());
export { Twilio };
var Mongo = (function () {
    function Mongo(name, type, config) {
        this.name = name;
        this.type = type;
        this.config = config;
    }
    return Mongo;
}());
export { Mongo };
