// mongodb-stitch/packages/core/admin-client/dist/umd/authProviders/AuthProvidersResources.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Fields;
    (function (Fields) {
        Fields["ID"] = "_id";
        Fields["DISABLED"] = "disabled";
        Fields["NAME"] = "name";
        Fields["TYPE"] = "type";
    })(Fields || (Fields = {}));
    var AuthProviderResponseCodec = (function () {
        function AuthProviderResponseCodec() {
        }
        AuthProviderResponseCodec.prototype.decode = function (from) {
            return {
                disabled: from[Fields.DISABLED],
                id: from[Fields.ID],
                name: from[Fields.NAME],
                type: from[Fields.TYPE]
            };
        };
        AuthProviderResponseCodec.prototype.encode = function (from) {
            var _a;
            return _a = {},
                _a[Fields.ID] = from.id,
                _a[Fields.DISABLED] = from.disabled,
                _a[Fields.NAME] = from.name,
                _a[Fields.TYPE] = from.type,
                _a;
        };
        return AuthProviderResponseCodec;
    }());
    exports.AuthProviderResponseCodec = AuthProviderResponseCodec;
});
