// mongodb-stitch/packages/core/services/aws-s3/dist/umd/internal/ResultDecoders.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var AwsS3PutObjectFields;
    (function (AwsS3PutObjectFields) {
        AwsS3PutObjectFields["Location"] = "location";
    })(AwsS3PutObjectFields || (AwsS3PutObjectFields = {}));
    var AwsS3PutObjectResultDecoder = (function () {
        function AwsS3PutObjectResultDecoder() {
        }
        AwsS3PutObjectResultDecoder.prototype.decode = function (from) {
            return {
                location: from[AwsS3PutObjectFields.Location]
            };
        };
        return AwsS3PutObjectResultDecoder;
    }());
    var AwsS3SignPolicyFields;
    (function (AwsS3SignPolicyFields) {
        AwsS3SignPolicyFields["Policy"] = "policy";
        AwsS3SignPolicyFields["Signature"] = "signature";
        AwsS3SignPolicyFields["Algoritm"] = "algorithm";
        AwsS3SignPolicyFields["Date"] = "date";
        AwsS3SignPolicyFields["Credential"] = "credential";
    })(AwsS3SignPolicyFields || (AwsS3SignPolicyFields = {}));
    var AwsS3SignPolicyResultDecoder = (function () {
        function AwsS3SignPolicyResultDecoder() {
        }
        AwsS3SignPolicyResultDecoder.prototype.decode = function (from) {
            return {
                algorithm: from[AwsS3SignPolicyFields.Algoritm],
                credential: from[AwsS3SignPolicyFields.Credential],
                date: from[AwsS3SignPolicyFields.Date],
                policy: from[AwsS3SignPolicyFields.Policy],
                signature: from[AwsS3SignPolicyFields.Signature]
            };
        };
        return AwsS3SignPolicyResultDecoder;
    }());
    var ResultDecoders = (function () {
        function ResultDecoders() {
        }
        ResultDecoders.PutObjectResultDecoder = new AwsS3PutObjectResultDecoder();
        ResultDecoders.SignPolicyResultDecoder = new AwsS3SignPolicyResultDecoder();
        return ResultDecoders;
    }());
    exports.default = ResultDecoders;
});
