// mongodb-stitch/packages/core/services/aws/dist/umd/AwsRequest.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var AwsRequest = (function () {
        function AwsRequest(service, action, region, args) {
            this.service = service;
            this.action = action;
            this.region = region;
            this.args = args;
        }
        return AwsRequest;
    }());
    exports.AwsRequest = AwsRequest;
    (function (AwsRequest) {
        var Builder = (function () {
            function Builder() {
            }
            Builder.prototype.withService = function (service) {
                this.service = service;
                return this;
            };
            Builder.prototype.withAction = function (action) {
                this.action = action;
                return this;
            };
            Builder.prototype.withRegion = function (region) {
                this.region = region;
                return this;
            };
            Builder.prototype.withArgs = function (args) {
                this.args = args;
                return this;
            };
            Builder.prototype.build = function () {
                if (!this.service || this.service == "") {
                    throw new Error("must set service");
                }
                if (!this.action || this.action == "") {
                    throw new Error("must set action");
                }
                return new AwsRequest(this.service, this.action, this.region, this.args || {});
            };
            return Builder;
        }());
        AwsRequest.Builder = Builder;
    })(AwsRequest = exports.AwsRequest || (exports.AwsRequest = {}));
    exports.AwsRequest = AwsRequest;
});
