// mongodb-stitch/packages/core/services/aws-ses/dist/umd/internal/CoreAwsSesServiceClient.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../AwsSesSendResult"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var AwsSesSendResult_1 = __importDefault(require("../AwsSesSendResult"));
    var CoreAwsSesServiceClient = (function () {
        function CoreAwsSesServiceClient(service) {
            this.service = service;
        }
        CoreAwsSesServiceClient.prototype.sendEmail = function (toAddress, fromAddress, subject, body) {
            var args = {
                body: body,
                fromAddress: fromAddress,
                subject: subject,
                toAddress: toAddress
            };
            return this.service.callFunction("send", [args], AwsSesSendResult_1.default.Decoder);
        };
        return CoreAwsSesServiceClient;
    }());
    exports.default = CoreAwsSesServiceClient;
});
