// mongodb-stitch/packages/core/admin-client/dist/umd/apps/AppsResources.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Fields;
    (function (Fields) {
        Fields["Id"] = "_id";
        Fields["Name"] = "name";
        Fields["ClientAppId"] = "client_app_id";
    })(Fields || (Fields = {}));
    var AppResponseCodec = (function () {
        function AppResponseCodec() {
        }
        AppResponseCodec.prototype.decode = function (from) {
            return {
                clientAppId: from[Fields.ClientAppId],
                id: from[Fields.Id],
                name: from[Fields.Name]
            };
        };
        AppResponseCodec.prototype.encode = function (from) {
            var _a;
            return _a = {},
                _a[Fields.Id] = from.id,
                _a[Fields.Name] = from.name,
                _a[Fields.ClientAppId] = from.clientAppId,
                _a;
        };
        return AppResponseCodec;
    }());
    exports.AppResponseCodec = AppResponseCodec;
});
