// mongodb-stitch/packages/core/services/twilio/dist/umd/CoreTwilioServiceClient.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var CoreTwilioServiceClient = (function () {
        function CoreTwilioServiceClient(service) {
            this.service = service;
        }
        CoreTwilioServiceClient.prototype.sendMessage = function (to, from, body, mediaUrl) {
            var args = {
                body: body,
                from: from,
                to: to
            };
            if (mediaUrl !== undefined) {
                args.mediaUrl = mediaUrl;
            }
            return this.service.callFunction("send", [args]);
        };
        return CoreTwilioServiceClient;
    }());
    exports.default = CoreTwilioServiceClient;
});
