// mongodb-stitch/packages/core/admin-client/dist/esm/StitchAdminUser.js
var StitchAdminUser = (function () {
    function StitchAdminUser(id, providerType, providerName, userProfile) {
        this.id = id;
        this.loggedInProviderType = providerType;
        this.loggedInProviderName = providerName;
        this.profile = userProfile;
    }
    Object.defineProperty(StitchAdminUser.prototype, "userType", {
        get: function () {
            return this.profile.userType;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(StitchAdminUser.prototype, "identities", {
        get: function () {
            return this.profile.identities;
        },
        enumerable: true,
        configurable: true
    });
    return StitchAdminUser;
}());
var StitchAdminUserFactory = (function () {
    function StitchAdminUserFactory() {
    }
    StitchAdminUserFactory.prototype.makeUser = function (id, loggedInProviderType, loggedInProviderName, userProfile) {
        return new StitchAdminUser(id, loggedInProviderType, loggedInProviderName, userProfile);
    };
    return StitchAdminUserFactory;
}());
export { StitchAdminUser, StitchAdminUserFactory };
