// mongodb-stitch/packages/core/admin-client/dist/umd/services/ServicesResources.js
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var Fields;
    (function (Fields) {
        Fields["Id"] = "_id";
        Fields["Name"] = "name";
        Fields["Type"] = "type";
    })(Fields || (Fields = {}));
    var ServiceResponseCodec = (function () {
        function ServiceResponseCodec() {
        }
        ServiceResponseCodec.prototype.decode = function (from) {
            return {
                id: from[Fields.Id],
                name: from[Fields.Name],
                type: from[Fields.Type]
            };
        };
        ServiceResponseCodec.prototype.encode = function (from) {
            var _a;
            return _a = {},
                _a[Fields.Id] = from.id,
                _a[Fields.Name] = from.name,
                _a[Fields.Type] = from.type,
                _a;
        };
        return ServiceResponseCodec;
    }());
    exports.ServiceResponseCodec = ServiceResponseCodec;
});
